registration/include/pcl/registration/correspondence_estimation_lookup_table.h

```diff
@@ -0,0 +1,517 @@
+        bool compute_distance_from_query_point_to_closest_point_;
+
+        /** \brief Number of queries performed on the lookup table. */
+        size_t number_of_queries_on_lookup_table_;
+
+        /** \brief Number of queries performed on the search tree (because they were outside the lookup table bounds). */
+        size_t number_of_queries_on_search_tree_;
+    };
+
+    /** \brief @b CorrespondenceEstimationLookupTable provides fast correspondence estimation
+      * by using a lookup table with pre-computed correspondences.
+      *
+      * \author Carlos M. Costa
+      * \ingroup registration
+      */
+    template <typename PointSource, typename PointTarget, typename Scalar = float>
+    class CorrespondenceEstimationLookupTable: public CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>
+    {
+      public:
+        typedef boost::shared_ptr<CorrespondenceEstimationLookupTable<PointSource, PointTarget, Scalar> > Ptr;
+        typedef boost::shared_ptr<const CorrespondenceEstimationLookupTable<PointSource, PointTarget, Scalar> > ConstPtr;
+
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::corr_name_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::force_no_recompute_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::force_no_recompute_reciprocal_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::indices_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::initCompute;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::initComputeReciprocal;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::input_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::target_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::tree_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::tree_reciprocal_;
+        using PCLBase<PointSource>::deinitCompute;
+
+        typedef pcl::search::KdTree<PointTarget> KdTree;
+        typedef typename pcl::search::KdTree<PointTarget>::Ptr KdTreePtr;
+
+        typedef pcl::search::KdTree<PointSource> KdTreeReciprocal;
+        typedef typename KdTree::Ptr KdTreeReciprocalPtr;
+
+        typedef pcl::PointCloud<PointSource> PointCloudSource;
+        typedef typename PointCloudSource::Ptr PointCloudSourcePtr;
+        typedef typename PointCloudSource::ConstPtr PointCloudSourceConstPtr;
+
+        typedef pcl::PointCloud<PointTarget> PointCloudTarget;
+        typedef typename PointCloudTarget::Ptr PointCloudTargetPtr;
+        typedef typename PointCloudTarget::ConstPtr PointCloudTargetConstPtr;
+
+        typedef typename KdTree::PointRepresentationConstPtr PointRepresentationConstPtr;
```

Comment:
You don't need to redefine all of these type again. CorrespondenceEstimationBase already has them. Use the keyword `using` instead.